File: src/main/java/br/com/blz/testjava/repository/ProductRepository.java
package br.com.blz.testjava.repository;

import br.com.blz.testjava.model.ProductEntity;

public interface ProductRepository {

    void createProduct(final ProductEntity productEntity);

    void updateProduct(final ProductEntity productModel);

    void deleteProduct(final Integer sku);

    ProductEntity findProductBySku(final Integer sku);
}

File: src/main/java/br/com/blz/testjava/repository/ProductRepository.kt
package br.com.blz.testjava.repository

import br.com.blz.testjava.model.Product

interface ProductRepository {

  fun save(entity: Product): Product?
  fun findBySku(sku: Long): Product
  fun findAll(): List<Product?>
  fun delete(entity: Product)
  fun deleteAll()
}
